scripts/ci_mingw64_geany.sh

```diff
@@ -0,0 +1,327 @@
+	echo "CC                   : ${CC}"
+	echo "CFLAGS               : ${CFLAGS}"
+	echo "Configure            : ${CONFIGURE_OPTIONS}"
+}
+
+
+patch_version_information() {
+	log "Patching version information"
+
+	if [ -z "${GEANY_GIT_REVISION}" ] && [ -z "${CI}" ]; then
+		return
+	fi
+
+	# parse version string and decrement the patch and/or minor levels to keep nightly build
+	# versions below the next release version
+	regex='^([0-9]*)[.]([0-9]*)([.]([0-9]*))?'
+	if [[ ${GEANY_VERSION} =~ $regex ]]; then
+		MAJOR="${BASH_REMATCH[1]}"
+		MINOR="${BASH_REMATCH[2]}"
+		PATCH="${BASH_REMATCH[4]}"
+		if [ -z "${PATCH}" ] || [ "${PATCH}" = "0" ]; then
+			MINOR="$((MINOR-1))"
+			PATCH="90"
+		else
+			PATCH="$((PATCH-1))"
+		fi
+	else
+		echo "Could not extract or parse version tag" >&2
+		exit 1
+	fi
+	# replace version information in configure.ac and for Windows binaries
+	sed -i -E "s/^AC_INIT.\[Geany\], \[(.+)\],/AC_INIT(\[Geany\], \[${GEANY_VERSION}\],/" ${GEANY_BUILD_DIR}/configure.ac
+	sed -i -E "s/^#define VER_FILEVERSION_STR[[:space:]]+\".*\"+/#define VER_FILEVERSION_STR \"${GEANY_VERSION}\"/" ${GEANY_BUILD_DIR}/geany_private.rc
+	sed -i -E "s/^#define VER_FILEVERSION[[:space:]]+[0-9,]+/#define VER_FILEVERSION ${MAJOR},${MINOR},${PATCH},90/" ${GEANY_BUILD_DIR}/geany_private.rc
+	sed -i -E "s/^[[:space:]]+version=\"[0-9.]+\"/version=\"${MAJOR}.${MINOR}.${PATCH}.90\"/" ${GEANY_BUILD_DIR}/geany.exe.manifest
+	sed -i -E "s/^!define PRODUCT_VERSION \"@VERSION@\"/!define PRODUCT_VERSION \"${GEANY_VERSION}\"/" ${GEANY_BUILD_DIR}/geany.nsi.in
+	sed -i -E "s/^!define PRODUCT_VERSION_ID \"@VERSION@.0.0\"/!define PRODUCT_VERSION_ID \"${MAJOR}.${MINOR}.${PATCH}.90\"/" ${GEANY_BUILD_DIR}/geany.nsi.in
+}
+
+
+build_geany() {
+	cd ${GEANY_BUILD_DIR}
+	log "Running autogen.sh"
+	./autogen.sh
+	log "Running configure"
+	./configure ${CONFIGURE_OPTIONS}
+	log "Running make"
+	make
+	log "Running install-strip"
+	make install-strip
+
+	cd /
+}
+
+
+sign_file() {
+	echo "Sign file $1"
+	if [ -f /certificates/cert.pem ] && [ -f /certificates/key.pem ]; then
+		osslsigncode sign \
+			-certs /certificates/cert.pem \
+			-key /certificates/key.pem \
+			-n "Geany Binary" \
+			-i "https://www.geany.org/" \
+			-ts http://zeitstempel.dfn.de/ \
+			-h sha512 \
+			-in ${1} \
+			-out ${1}-signed
+		mv ${1}-signed ${1}
+	else
+		echo "Skip signing due to missing certificate"
+	fi
+}
+
+
+sign_geany_binaries() {
+	log "Signing Geany binary files"
+	for binary_file_pattern in ${GEANY_RELEASE_BINARY_PATTERNS[@]}; do
+		for binary_file in $(ls ${binary_file_pattern}); do
+			sign_file ${binary_file}
+		done
+	done
+}
+
+
+convert_text_files_to_crlf() {
+	log "Converting line endings to CRLF in text files"
+	for text_file_pattern in ${GEANY_RELEASE_TEXTFILE_PATTERNS[@]}; do
+		for text_file in $(ls ${text_file_pattern}); do
+			mime_type=$(file --brief --mime-type ${text_file})
+			case $mime_type in text/*)
+				unix2dos ${text_file}
```

Comment:
Isn't that something that should be done in geany build itself? Anyone who builds Geany from source on Windows should do this right?

Reply:
Git should convert line endings on checkout since the default value of `core.eol` is `native` but maybe MSYS2 doesn't register "native" as CRLF?  Maybe the scripts in MSYS2 should do `git config core.eol crlf`?

Reply:
I don't expect from Geany's build system to know whether it need to convert text files to the target system's preferences.
IMO this is very Windows specific and basically only necessary to be able to read our text files with Windows' Notepad which is not able to handle different line endings.

In Geany, we would need to properly integrate it in Autotools and Meson and also depend on "unix2dos".

Regarding Git's `core.eol`: I'm not what happens on a native Windows clone but here we use the already cloned repository from the CI job and even if we would clone it ourselves, it would be LF as everything in this script is executed on Linux.

Changing the setting to `crlf` might have other side effects we do not want.

Reply:
> Windows' Notepad which is not able to handle different line endings.

Works on my windows 11 tablet[^1].  Google says it was fixed in 2018, so any  _supported_ version of Windows should be ok ;-P

> Regarding Git's core.eol: I'm not what happens on a native Windows clone but here we use the already cloned repository from the CI job and even if we would clone it ourselves

Ok, so its necessary for the CI job, but I was really addressing @kugel- "Anyone who builds Geany from source on Windows should do this right?" but IIUC their checkout would do it for them with default git settings.  If tarballs are going to be buildable on windows I guess its needed there as well since the tarball files are LF endings unless we make a special Windows tarball ... or just say they are *x or modern Windows only.

> Changing the setting to crlf might have other side effects we do not want.

Yes, IIUC "Bad things" will happen if we merge anything back on Linux with that setting.

[^1]: No, its a _tablet_ so Geany is no use as I don't develop on it, sorry can't help with windows builds ;-P

Reply:
Well the CI builds shouldn't have "more features" than normal builds from source - source code is king!! If a hard dependency in the normal build is not wanted then there could be an optional dependency that is _coincidentally_ met by the CI.

I wouldn't want manual git or tarball builds to regress compared to CI builds (e.g. on Windows 7). Stuff like this is hard to track down even for experienced contributors.

IMO: Let's open an issue regarding the line ending issue. Then we either deal with it in the normal build or just live with it as suggested by @elextr . I can help with meson if that's a problem.

Reply:
> IMO: Let's open an issue regarding the line ending issue. 

Agreed, and I would have thought the tarball problem would exist already, can anyone test it?

Reply:
Could any of you open that issue?
I still don't get the problem you see, as outlined above, a GIT clone on Windows probably gives the correct line endings and so I'm not really able to create an issue for an issue I don't understand :).

Here we are talking about a native Linux clone with native LF line endings and those we need to convert. IMO this is special to the CI build and so needs to be done here.

Reply:
A tarball created on Linux will have wrong line endings on Windows. The normal build system should take care of that instead of being restricted to CI.

And how do you create Win32 releases if not using CI (as you mentioned in another comment)?

Reply:
> A tarball created on Linux will have wrong line endings on Windows. The normal build system should take care of that instead of being restricted to CI.

As said above, if you want to do this, please create an issue or even better a PR to make the changes to the build system(s). Then we can remove the line conversion here.
I won't do it because I still think it's not such a big deal and I won't spend the time effort it needs.

> And how do you create Win32 releases if not using CI (as you mentioned in another comment)?

As described on https://wiki.geany.org/howtos/win32/msys2, especially using the [attached](https://wiki.geany.org/_media/howtos/win32/geany-release.py.txt) script which also does the line ending conversion to CRLF.
In short, using a clean GIT clone with the release tag. At least with my GIT config the relevant files all have LF line endings and therefore need to be converted.
Using a release tarball wouldn't help here because it is created on Linux and so also has LF line endings.

Reply:
I opened #3350 

I would like to see the conversion to be removed from this PR since I'm not even sure it's necessary at all. Don't recent Windows version deal fine with LF line endings. I don't know if notepad alone is an indication but see https://devblogs.microsoft.com/commandline/extended-eol-in-notepad/, this fix is included in Windows 10 and 11.

Reply:
Thanks for creating the issue and the investigation.
Nice that Notepad now can handle non-CRLF line endings. This was the main reason I convert the line endings.

So I just removed the code from the build script here and in the G-P counterpart.